src/overlays/actors/ovl_En_Tab/z_en_tab.c

```diff
@@ -31,75 +65,552 @@ const ActorInit En_Tab_InitVars = {
+    this->unk_314 = CLAMP(this->unk_314, -0x1FFE, 0x1FFE);
+
+    Math_ApproachS(&this->unk_318, (sp32 - this->unk_314) - this->actor.shape.rot.y, 4, 0x2AA8);
+    this->unk_318 = CLAMP(this->unk_318, -0x1C70, 0x1C70);
+
+    Math_Vec3f_Copy(&sp34, &this->actor.focus.pos);
+    if (this->unk_1E0->id == ACTOR_PLAYER) {
+        sp40.y = ((Player*)this->unk_1E0)->bodyPartsPos[7].y + 3.0f;
+    } else {
+        Math_Vec3f_Copy(&sp40, &this->unk_1E0->focus.pos);
+    }
+
+    Math_ApproachS(&this->unk_312, Math_Vec3f_Pitch(&sp34, &sp40) - this->unk_316, 4, 0x2AA8);
+    this->unk_312 = CLAMP(this->unk_312, -0x1554, 0x1554);
+
+    Math_ApproachS(&this->unk_316, Math_Vec3f_Pitch(&sp34, &sp40) - this->unk_312, 4, 0x2AA8);
+    this->unk_316 = CLAMP(this->unk_316, -0xE38, 0xE38);
+}
+
+void func_80BE09A8(EnTab* this, GlobalContext* globalCtx) {
+    this->unk_1E0 = func_80BE0778(this, globalCtx);
+    if ((this->unk_2FC & 8) && (this->unk_1E0 != 0) && (DECR(this->unk_324) == 0)) {
+        func_80BE07A0(this);
+        this->unk_316 = 0;
+        this->unk_318 = 0;
+        this->unk_2FC &= ~0x40;
+        this->unk_2FC |= 0x10;
+    } else if (this->unk_2FC & 0x10) {
+        this->unk_2FC &= ~0x10;
+        this->unk_312 = 0;
+        this->unk_314 = 0;
+        this->unk_316 = 0;
+        this->unk_318 = 0;
+        this->unk_324 = 20;
+    } else if (DECR(this->unk_324) == 0) {
+        this->unk_2FC |= 0x40;
+    }
+}
+
+void func_80BE0A98(EnTab* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s32 sp20 = Message_GetState(&globalCtx->msgCtx);
+
+    this->unk_308 += (this->unk_304 != 0.0f) ? 40.0f : -40.0f;
+    this->unk_308 = CLAMP(this->unk_308, 0.0f, 80.0f);
+
+    Matrix_InsertTranslation(this->unk_308, 0.0f, 0.0f, MTXMODE_APPLY);
+
+    if ((&this->actor == player->targetActor) &&
+        ((globalCtx->msgCtx.unk11F04 < 0xFF) || (globalCtx->msgCtx.unk11F04 > 0x200)) && (sp20 == 3) &&
+        (this->unk_334 == 3)) {
+        if ((globalCtx->state.frames % 2) == 0) {
+            if (this->unk_304 != 0.0f) {
+                this->unk_304 = 0.0f;
+            } else {
+                this->unk_304 = 1.0f;
+            }
+        }
+    } else {
+        this->unk_304 = 0.0f;
+    }
+    this->unk_334 = sp20;
+}
+
+s32 func_80BE0C04(EnTab* this, Actor* actor, f32 arg2) {
+    Vec3f sp44;
+    Vec3f sp38;
+    s32 ret = false;
+    f32 sp30;
+    s16 sp2E;
+
+    if (actor != NULL) {
+        Math_Vec3f_Copy(&sp38, &this->actor.world.pos);
+        Math_Vec3f_Copy(&sp44, &actor->world.pos);
+        sp2E = Math_Vec3f_Yaw(&sp38, &sp44);
+
+        if (this->unk_338 != 0) {
+            sp30 = arg2;
+        } else {
+            sp30 = arg2 * 0.5f;
+        }
+        if ((Math_Vec3f_DistXZ(&sp44, &sp38) < sp30) && (ABS_ALT(BINANG_SUB(sp2E, this->actor.shape.rot.y)) < 0x38E0)) {
+            ret = true;
+        }
+
+        if (DECR(this->unk_322) == 0) {
+            this->unk_338 ^= 1;
```

Comment:
```
        if (this->unk_338 != 0) {
...
            this->unk_338 ^= 1;
```
Hmm, if we're XORing against a static 1 this is a toggle? Should we treat it as boolean and use 
```
        if (this->unk_338) {
```

Reply:
Looks like this value is only used as a boolean, so it makes sense.